### PR TITLE
#56 current conversation bugfix

### PR DESCRIPTION
**Pull request by camelPhonso**
Updated the `useEffect` in CurrentConversation to set the currentConversation to the first in the array on initial render.

There was a suggestion to do this as a lazy initialisation but that would be setting the state during render and when tested seems to counteract the effect of our `updateOpenConvo` handler being passed from `ConversationsList`

### CODE REVIEW

File: components/messaging/CurrentConversation.tsx [hunk 1]
```diff
@@ -13,25 +13,16 @@ const CurrentConversation: React.FC = () => {
...
-
-    setCurrentConversation &&
-      setCurrentConversation(
-        allConversations?.filter(
-          (conversation) => conversation.conversation_id === highestId
-        )[0]
-      );
-  }, []);
+    setCurrentConversation && setCurrentConversation(allConversations[0]);
+  }, [allConversations, setCurrentConversation]);
```
Comment (nichgalzin) [inline components/messaging/CurrentConversation.tsx:18]: I have the feeling this is maybe not correct. Why do we want these dependency? We don't want this use effect to run when these things change, as that would reset the current convo when a new conversation is added or anytime a user clicked to open a new conversation. I think we just want this run once on mount. So have no dependencies.

File: components/messaging/CurrentConversation.tsx [hunk 2]
```diff
@@ -41,7 +32,7 @@ const CurrentConversation: React.FC = () => {
     };
 
     fetchMessagesForCurrentConversation();
-  }, [currentConversation]);
+  }, [currentConversation, setCurrentMessages]);
```
Comment (nichgalzin) [inline components/messaging/CurrentConversation.tsx:36]: Not sure we need this new dependency either, we don't need all the messages fetched when  setcurrentMessages runs, that's what the supabase reatime is managing.

Comment (camelPhonso) [reply components/messaging/CurrentConversation.tsx:36]: Marking this as resolved following our call :)

File: components/messaging/CurrentConversation.tsx (hunk 1 above)
Comment (camelPhonso) [reply components/messaging/CurrentConversation.tsx:18]: Marking this as resolved following our conversation :)

**Pull request merged**